sql/mozfun/json/from_nested_map/udf.sql

```diff
@@ -0,0 +1,28 @@
+-- Definition for json.from_nested_map
+CREATE OR REPLACE function json.from_nested_map(input JSON)
+RETURNS json
+LANGUAGE js
+AS
+  """
+  if (input && Object.keys(input).length) {
+    for (const k in input) {
+      if (input[k] && input[k].length) {
+        input[k] = input[k].reduce((acc, {key, value}) => {
+          acc[key] = value;
+          return acc;
+        }, {});
+      }
+    }
+    return input;
+  }
+  return null;
+""";
+
+-- Tests
+SELECT
+  assert.null(json.from_nested_map(NULL)),
+  assert.json_equal(JSON '{ "metric": {} }', json.from_nested_map(JSON '{ "metric": [] }')),
```

Comment:
I would suggest adding a few more test cases.

For example what would be the expected out come for the following: 
```json
{"metric": [1, 2, 3]}
```
```json
{"metric": "value"}
```
```json
{"metric": [{"key": "value"}]}
```